lib/beetle/publisher.rb

```diff
@@ -2,6 +2,9 @@ module Beetle
   # Provides the publishing logic implementation.
   class Publisher < Base
 
+    DEFAULT_BUNNY_CONNECT_TIMEOUT = 5
+    DEFAULT_BEETLE_DISCONNECT_TIMEOUT = DEFAULT_BUNNY_CONNECT_TIMEOUT * 2 + 1
```

Comment:
why not just the same as the bunny connect timeout?

Reply:
Because bunny applies the disconnect timeout twice:

```ruby
# from https://github.com/ruby-amqp/bunny/blob/v0.7.11/lib/qrack/client.rb#L77-L99
# Close all active channels
channels.each do |c|
  Bunny::Timer::timeout(@disconnect_timeout) { c.close } if c.open?
end

# Close connection to AMQP server
Bunny::Timer::timeout(@disconnect_timeout) { close_connection }
```

Reply:
Actually, I think we should change the bunny code to use the normal read_write_timeout for closing the channels and the connect timeout for the final close of the connection.

Reply:
Which in our case would mean no timeout (as it defaults to 0).
This leaves the question to what we want to set the timeout within beetle? `socket_timeout + connect_timeout + 1`?

Reply:
Seems like the right compromise to me. But first we need to patch bunny.

Reply:
A single `@disconnect_timeout`-based timeout around closing all channels and the connection would be what I as a user of the library would expect, but I see that 
https://github.com/ruby-amqp/bunny/pull/576 already got merged 😉

Reply:
I can provide some justification here. Closing a channel is a "regular" protocol operation in my [implement's] eyes, so "regular" channel timeouts should apply. However, there's only one disconnection operation, so a special timeout for it is fine. Note that a connection doesn't have to close all of its channels first, it's merely an internal housekeeping feature that many clients adopted.

Reply:
Thy @michaelklishin for weighing in and also releasing bunny version 0.7.12!